internal/exec/exec_test.go

```diff
@@ -0,0 +1,64 @@
+package exec
+
+import (
+	"os"
+	"testing"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"golang.org/x/sys/windows"
+)
+
+func TestExecProcess(t *testing.T) {
+	var procThreadSize uintptr
+	if err := winapi.InitializeProcThreadAttributeList(nil, 1, 0, &procThreadSize); err != nil && err != windows.E_NOT_SUFFICIENT_BUFFER {
+		t.Fatalf("failed to call InitializeProcThreadAttributeList: %s", err)
+	}
+
+	procHeap, err := winapi.GetProcessHeap()
+	if err != nil {
+		t.Fatalf("GetProcessHeap failed: %s", err)
+	}
+
+	attrList, err := winapi.HeapAlloc(procHeap, 0, procThreadSize)
+	if err != nil {
+		t.Fatalf("HeapAlloc failed: %v\n", err)
+	}
+	defer winapi.HeapFree(procHeap, 0, attrList)
+
+	var procAttrList *winapi.PROC_THREAD_ATTRIBUTE_LIST
+	procAttrList = (*winapi.PROC_THREAD_ATTRIBUTE_LIST)(unsafe.Pointer(attrList))
+	if err = winapi.InitializeProcThreadAttributeList(procAttrList, 1, 0, &procThreadSize); err != nil {
+		t.Fatalf("InitializeProcThreadAttributeList - second call failed: %v\n", err)
+	}
+
+	// This is useless but it tests all the win32 calls working. A better use for this test would be creating
+	// this process to be a child of a different process other than the one launching it but we can't gurantee
+	// any other processes running/get their pids.
+	pHandle, err := windows.OpenProcess(winapi.PROCESS_ALL_ACCESS, false, uint32(os.Getpid()))
+	uintPtrHandle := uintptr(pHandle)
+	if err := winapi.UpdateProcThreadAttribute(
+		procAttrList,
+		0,
+		winapi.PROC_THREAD_ATTRIBUTE_PARENT_PROCESS,
+		&uintPtrHandle,
+		unsafe.Sizeof(pHandle),
+		0,
+		nil,
+	); err != nil {
+		t.Fatalf("UpdateProcThreadAttrite failed: %s", err)
+	}
+	defer winapi.DeleteProcThreadAttributeList(procAttrList)
```

Comment:
why isn't this defer after the creation of procAttrList?

Reply:
Likely an oversight 😋